docs/docs/directives/schema-directives.md

```diff
@@ -0,0 +1,133 @@
+
+## Built-In Directives
+
+The GraphQL spec defines directives that are supported out of the box in EntityGraphQL.
+
+- `@deprecated(reason: String)` - Tells the client that this type, field or enum value should no longer be used along with the reason why or a suggested alternative
+- `@oneOf` - Skip this field if the argument is true.
+- `@specifiedBy(url: String)` - Skip this field if the argument is true.
+
+## Deprecated
+
+Fields, Enum Values, Mutation Arguments and Types marked with the c# [ObsoleteAttribute] will have the [@deprecated](https://spec.graphql.org/draft/#sec--deprecated) directive automatically added.
+
+You can also use the Deprecate(string reason) extension method on a IField.
+
+## One Of Input Types
+
+EntityGraphQL supports [One Of Input Types](https://github.com/graphql/graphql-spec/pull/825).
+
+Mark an input type with `GraphQLOneOfAttribute` and EntityGraphQL will mark the type with `@oneOf` in the schema and validate the input meets the requiements on execution.
+
+```cs
+[GraphQLOneOf]
+private class OneOfInputType
+{
+    public int? One { get; set; }
+    public int? Two { get; set; }
+}
+```
+
+This will generate the follow graphql schema.
+
+```graphql
+input MutationArgs @oneOf {
+  one: Int
+  two: Int
+}
+```
+
+Although each field on the input is nullable the `@oneOf` input type has one further validation step.
+
+- Exactly one key must be specified
+
+## Specified By
+
+The [@specifiedBy](https://spec.graphql.org/draft/#sec--specifiedBy) directive is used to provide extra information about a custom scalar type.
+
+## Custom Directives
+
+Like [Field Extensions](../field-extensions/) you can extend the [ExtensionAttribute](../other-extensibility/extension-attribute) to apply extension to `IFields` and `ISchemaTypes` (see `GraphQLOneOfAttribute`).
+
+IFields and ISchema types also have a method `AddDirective` that is common to call in the attribute extension method or a custom extension method to register the directive.
+
+```
+public static class DeprecatedDirectiveExtensions
+{
+    public static void Deprecate(this IField field, string? reason)
+    {
+        field.AddDirective(new DeprecatedDirective(reason));
+    }
+}
+public class ObsoleteAttributeRegistration : IExtensionAttribute<ObsoleteAttribute>
+{
+    public void ApplyExtension(IField field, ObsoleteAttribute attribute)
+    {
+        field.Deprecate(attribute.Message);
+    }
+}
+```
+
+The `ISchemaDirective` method contains:
+
+* The `On` property for marking where it is valid to use this directive
+* The `ToGraphQLSchemaString()` method where you return the formatted directive including arguments (eg `return $"@deprecated(reason: \"{Reason}\")";`)
+* A series of methods the SchemaInstrospection class calls allowing you to modify the introspection result (primarily used because instrospection for @deprecated and @oneOf is done using specific fields and not yet in an extensible way)
```

Comment:
Can you explain this comment more please?

Reply:
I mainly meant the "not yet in an extensible way" part. Which bit is not yet extensible? 

It says `@deprecated` is the original comment but you mention `@deprecated` has a "nice reusable syntax" above?

Just trying to understand what this means? Each directive seemed to implement their own functionality, am I missing something?

Reply:
"not yet in an extensible way" as in there's no official place to put information around new directives you make.  You have ot invent a new field and look it up yourself.

The @deprecated comment was more about the above sentence eg SDL @directive syntax supports any directive, but that information isn't officially exposed in introspection for custom directives.

Reply:
should probably be changed to a sentence stating something like

'directives are automatically added to the SDL output. As there is no official way to view directives in the introspection spec as of yet, the recommend method is extending the built in introspection types with additional fields with the information you wish to expose.

eg the @oneOf directive is exposed as the field OneField on TypeElement"

Reply:
Oh I get it now.

Umm yeh interesting. `OneField` is nearly in spec. hopefully they bring in directive introspection into spec too.

Your above sentence is more to the point I think.